airbyte-integrations/connectors/source-stripe/source_stripe/streams.py

```diff
@@ -55,9 +55,15 @@ def __init__(self, cursor_field: str, legacy_cursor_field: Optional[str]):
     def extract_records(self, response: requests.Response) -> Iterable[MutableMapping]:
         records = super().extract_records(response)
         for record in records:
-            if self.legacy_cursor_field and self.cursor_field not in record:
-                record[self.cursor_field] = record[self.legacy_cursor_field]
-            yield record
+            if self.cursor_field in record:
+                yield record
+                continue  # Skip the rest of the loop iteration
+
+            # fetch legacy_cursor_field from record; default to current timestamp for initial syncs without an any cursor.
+            current_cursor_value = record.get(self.legacy_cursor_field, pendulum.now().int_timestamp)
+
+            # yield the record with the added cursor_field
+            yield record | {self.cursor_field: current_cursor_value}
```

Comment:
Defaulting on `now` seems odd to me. Is it only some records in a stream that might not have `self.cursor_field` and `self.legacy_cursor_field` in the record? I would be afraid of the following series of events:
* Consume a record #&NoBreak;1 `self.cursor_field` and `self.legacy_cursor_field` not in the record so we use `now`
* checkpointing from AbstractSource
* Consume a record #&NoBreak;2 with `self.cursor_field` or `self.legacy_cursor_field` less than `now`
* Sync crash

In that case, we would start the next sync at the `now` when we consumed record #&NoBreak;1 but would potentially miss record #&NoBreak;2 and others. The only reason allow for `now` as the default is if all the records for the same stream will not have `self.legacy_cursor_field` as a field. Can you confirm?

Reply:
@maxi297 yes, the case here is *some* streams that do not have a cursor value at all when running in a full refresh mode (or initial incremental sync), so the consequences you described are not relevant